tests/rptest/services/redpanda.py

```diff
@@ -37,6 +38,58 @@
 MetricSample = collections.namedtuple(
     'MetricSample', ['family', 'sample', 'node', 'value', 'labels'])
 
+DEFAULT_LOG_ALLOW_LIST = [
+    # Tests currently don't run on XFS, although in future they should.
+    # https://github.com/vectorizedio/redpanda/issues/2376
+    re.compile("not on XFS. This is a non-supported setup."),
+
+    # This is expected when tests are intentionally run on low memory configurations
+    re.compile(r"Memory: '\d+' below recommended"),
+    # A client disconnecting is not bad behaviour on redpanda's part
+    re.compile(r"kafka rpc protocol.*(Connection reset by peer|Broken pipe)")
+]
+
+# Log errors that are expected in tests that restart nodes mid-test
+RESTART_LOG_ALLOW_LIST = [
+    re.compile("(raft|rpc) - .*(disconnected_endpoint)"),
+]
+
+# Log errors that are expected in chaos-style tests that e.g.
+# stop redpanda nodes uncleanly
+CHAOS_LOG_ALLOW_LIST = [
+    # Unclean connection shutdown
+    re.compile(
+        "(raft|rpc) - .*(client_request_timeout|disconnected_endpoint|Broken pipe|Connection reset by peer)"
+    ),
+
+    # Torn disk writes
+    re.compile("storage - Could not parse header"),
+
+    # e.g. raft - [group_id:59, {kafka/test-topic-319-1639161306093460/0}] consensus.cc:2301 - unable to replicate updated configuration: raft::errc::replicated_entry_truncated
+    re.compile("raft - .*replicated_entry_truncated"),
+
+    # e.g. cluster - controller_backend.cc:466 - exception while executing partition operation: {type: update_finished, ntp: {kafka/test-topic-1944-1639161306808363/1}, offset: 413, new_assignment: { id: 1, group_id: 65, replicas: {{node_id: 3, shard: 2}, {node_id: 4, shard: 2}, {node_id: 1, shard: 0}} }, previous_assignment: {nullopt}} - std::__1::__fs::filesystem::filesystem_error (error system:39, filesystem error: remove failed: Directory not empty [/var/lib/redpanda/data/kafka/test-topic-1944-1639161306808363])
+    re.compile("cluster - .*Directory not empty"),
```

Comment:
@dotnwat can you sanity check this Directory not empty one?  This comes up during node_operations_fuzzy_test, and I'm presuming it's a race between the storage layer tearing down a partition and the controller trying to shut it down, but if it's not meant to work that way this could be a bug.

Reply:
I'd be inclined to try to fix instances of this because it could be all sorts of things like forgetting to clean-up some tmp files in partitions, but that may not apply to this particular fuzzy_test and it should be on that test's allow list.

I'll try to reproduce and see what's happening.

Reply:
From a completely different thread, this could be the bug where snapshots aren't deleted https://github.com/vectorizedio/redpanda/issues/3287

Reply:
ahh! good catch. I added #3287 to the queue for tomorrow. should be something good for Friday.